plugins/janus_streaming.c

```diff
@@ -6868,6 +6877,15 @@ janus_streaming_mountpoint *janus_streaming_create_rtsp_source(
 	live_rtsp->codecs.audio_pt = doaudio ? acodec : -1;
 	live_rtsp->codecs.audio_rtpmap = doaudio ? (artpmap ? g_strdup(artpmap) : NULL) : NULL;
 	live_rtsp->codecs.audio_fmtp = doaudio ? (afmtp ? g_strdup(afmtp) : NULL) : NULL;
+	live_rtsp->codecs.video_codec = JANUS_VIDEOCODEC_NONE;
+	if(dovideo && vrtpmap) {
+		if(strstr(vrtpmap, "vp8") || strstr(vrtpmap, "VP8"))
+			live_rtsp->codecs.video_codec = JANUS_VIDEOCODEC_VP8;
+		else if(strstr(vrtpmap, "vp9") || strstr(vrtpmap, "VP9"))
+			live_rtsp->codecs.video_codec = JANUS_VIDEOCODEC_VP9;
+		else if(strstr(vrtpmap, "h264") || strstr(vrtpmap, "H264"))
+			live_rtsp->codecs.video_codec = JANUS_VIDEOCODEC_H264;
+	}
```

Comment:
Not sure about this. Normally with RTSP you just say if you want audio and video, you don't specify the RTP map too. That's because it all depends on what the RTSP server will be negotiated, meaning it will be dynamic. This should be done *after* the session has been negotiated, not before.